benefits/core/session.py

```diff
@@ -137,6 +145,7 @@ def reset(request):
     """Reset the session for the request."""
     logger.debug("Reset session")
     request.session[_AGENCY] = None
+    request.session[_AUTH] = None
```

Comment:
Here we may want to default to `False`, to be explicit that on session reset the user is not signed in.